File: contest/typical90AD/typical90AD.py
import sys

input = lambda: sys.stdin.readline().rstrip()
N, K = map(int, input().split())


MAXN = N + 1
sieve = [i for i in range(MAXN)]  # sieve[i]: iの最も小さい素因数
count = [0] * MAXN
count[1] = 1
p = 2
while p * p <= MAXN:  # O(MAXN * log MAXN)
    if sieve[p] == p:
        for q in range(p, MAXN, p):
            if sieve[q] == q:
                sieve[q] = p
            count[q] += 1
    p += 1


ans = sum([c >= K for c in count])

print(ans)


# osa_k法
MAXN = 10**6 + 10
sieve = [i for i in range(MAXN)]  # sieve[i]: iの最も小さい素因数
p = 2
while p * p <= MAXN:  # O(MAXN * loglog MAXN)
    if sieve[p] == p:
        for q in range(p * p, MAXN, p):
            if sieve[q] == q:
                sieve[q] = p
    p += 1

from collections import defaultdict


def prime_factors(n):
    """n(2<=n<=MAXN)の素因数を返す"""
    d = defaultdict(int)  # 適宜変更
    while n > 1:  # O(log n)
        d[sieve[n]] += 1
        n //= sieve[n]
    return d

File: contest/typical90AD/typical90AD.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;

ll MAXN = 10000010;
ll sieve[10000010];
ll N, K;

void osaK(){
    for (ll p=2; p*p<MAXN; p++){
        if (sieve[p]!=p) continue;

        for (ll q=p*p; q<MAXN; q+=p){
            if (sieve[q]!=q) continue;

            sieve[q] = p;
        }
    }
}

map<ll, int> primeFactors(ll n){
    if (n==0) return {};
    if (n==1) return {{1, 1}};
    map<ll, int> rev;
    while (n>1)
    {
        rev[sieve[n]]++;
        n /= sieve[n];
    }
    return rev;
}

int main(){
    cin >> N >> K;
    for (ll i=0; i<MAXN; i++) sieve[i] = i;
    osaK();
    ll ans = 0;
    for (ll i=2; i<=N; i++){
        if (primeFactors(i).size()>=K) ans++;
    }
    cout << ans << endl;
    return 0;
}